demisto_sdk/commands/download/downloader.py

```diff
@@ -1,741 +1,978 @@
+        logger.debug(
+            f"Downloaded content bundle size (bytes): {len(api_response.data)}"
+        )
+
+        loaded_files: dict[str, StringIO] = {}
+
+        with tarfile.open(fileobj=BytesIO(api_response.data), mode="r") as tar:
+            tar_members = tar.getmembers()
+            logger.debug(f"Custom content bundle contains {len(tar_members)} items.")
+
+            for file in tar_members:
+                file_name = file.name.lstrip("/")
+
+                if extracted_file := tar.extractfile(file):
+                    file_data = StringIO(safe_read_unicode(extracted_file.read()))
+                    loaded_files[file_name] = file_data
+
+        logger.debug("Custom content items loaded to memory successfully.")
+        return loaded_files
+
+    def replace_uuid_ids(
+        self, custom_content_object: dict, uuid_mapping: dict[str, str]
+    ) -> bool:
         """
-        Fetches the custom content from Demisto into a temporary dir.
-        :return: True if fetched successfully, False otherwise
+        Find and replace UUID IDs of custom content items with their names.
+        The method first creates a mapping of a UUID to a name, and then replaces all UUIDs using this mapping.
+
+        Args:
+            custom_content_object (dict): A single custom content object to update UUIDs in.
+            uuid_mapping (dict[str, str]): A dictionary mapping UUID IDs to corresponding names of custom content.
+
+        Returns:
+            bool: True if the object was updated, False otherwise.
         """
-        try:
-            verify = (
-                (not self.insecure) if self.insecure else None
-            )  # set to None so demisto_client will use env var DEMISTO_VERIFY_SSL
-            self.client = demisto_client.configure(verify_ssl=verify)
-            api_response: tuple = demisto_client.generic_request_func(
-                self.client, "/content/bundle", "GET"
-            )
-            body: bytes = ast.literal_eval(api_response[0])
-            io_bytes = io.BytesIO(body)
-
-            # Demisto's custom content file is of type tar.gz
-            tar = tarfile.open(fileobj=io_bytes, mode="r")
-
-            (
-                content_items_file_names_tuple,
-                scripts_id_to_name,
-            ) = self.find_uuids_in_content_item(tar)
-            self.write_custom_content(
-                content_items_file_names_tuple, scripts_id_to_name
+        content_item_file_str = custom_content_object["file"].getvalue()
+        uuid_matches = re.findall(UUID_REGEX, content_item_file_str)
+
+        if uuid_matches:
+            for uuid in set(uuid_matches).intersection(uuid_mapping):
+                logger.debug(
+                    f"Replacing UUID '{uuid}' with '{uuid_mapping[uuid]}' in "
+                    f"'{custom_content_object['name']}'"
+                )
+                content_item_file_str = content_item_file_str.replace(
```

Comment:
```suggestion
                content_item_file_name = content_item_file_str.replace(
```
its not just the file name?

Reply:
It's the full file represented as a string (for example, the whole playbook's YAML).
```python
content_item_file_str = custom_content_object["file"].getvalue()
```
(`custom_content_object["file"]` is a `StringIO` representation of the file.)

We load the whole file as a string and then lookup for UUIDs within it using a RegEx.